src/main/java/gov/nasa/cumulus/metadata/aggregator/MetadataFilesToEcho.java

```diff
@@ -542,14 +546,26 @@ public IsoGranule createIsoCyclePassTile(String cyclePassTileStr) {
 		}
 		TrackType trackType = null;
 		List<AdditionalAttributeType> additionalAttributeTypes = new ArrayList<>();
+		/**
+		 * This block of code supports multiple cycles.  In theory, during cycle transition, it is possible
+		 * a granule consists 2 cycles.  However, UMMG json schema does support one at the time.
+		 */
 		for(String cps : cyclePassStrs) {
-			trackType = createTrackType(cps, p_cycle);
+			try {
+				trackType = createTrackType(cps, p_cycle);
+			} catch (Exception e) {
+				AdapterLogger.LogWarning(this.className + " continue processing after creating TrackType with exception: " + UMMUtils.getStackTraceAsString(e));
+			}
 			UmmgPojoFactory ummgPojoFactory = UmmgPojoFactory.getInstance();
 			additionalAttributeTypes=
 					ummgPojoFactory.trackTypeToAdditionalAttributeTypes(trackType);
 		}
-		((IsoGranule)granule).setTrackType(trackType);
-		((IsoGranule)granule).setAdditionalAttributeTypes(additionalAttributeTypes);
+		// It is possible after all the above processing, cycle is present but passes is not (no pass in passes array)
+		// That is, we shall NOT create trackType at all.  Otherwise, CMR will throw validation error
+		if (trackType.getCycle()!=null && trackType.getPasses()!=null && trackType.getPasses().size() >0) {
```

Comment:
Can't this throw a NPE if trackType isn't set (meaning we hit the catch block on line 557)?

Reply:
Scratch that, I see we have a try/catch around the `createIsoCyclePassTile` call